crates/nexmark/examples/generate.rs

```diff
@@ -0,0 +1,157 @@
+use std::mem::replace;
+
+use clap::Parser;
+use csv::{Writer, WriterBuilder};
+use dbsp_nexmark::{config::GeneratorOptions, NexmarkSource};
+use env_logger::Env;
+use indicatif::ProgressBar;
+use rdkafka::{
+    config::FromClientConfig,
+    producer::{base_producer::ThreadedProducer, BaseRecord, DefaultProducerContext, Producer},
+    util::Timeout,
+    ClientConfig,
+};
+use serde::Serialize;
+
+/// Accumulates data until it crosses a threshold size and then outputs it to it
+/// to a Kafka topic.
+struct BufferedTopic<'a> {
+    producer: &'a ThreadedProducer<DefaultProducerContext>,
+    topic: String,
+    writer: Writer<Vec<u8>>,
+}
+
+impl<'a> BufferedTopic<'a> {
+    /// Creates a new `BufferedTopic` to output to `topic` via `producer`.
+    pub fn new(producer: &'a ThreadedProducer<DefaultProducerContext>, topic: String) -> Self {
+        Self {
+            producer,
+            topic,
+            writer: Self::new_writer(),
+        }
+    }
+
+    /// Writes `record`, flushing automatically if the threshold is crossed.
+    pub fn write<S>(&mut self, record: S)
+    where
+        S: Serialize,
+    {
+        self.writer.serialize(record).unwrap();
+        self.flush_if(512 * 1024);
```

Comment:
should this be a more visible constant?

Reply:
I added a `MAX_LEN` constant and made the threshold more precise.